### ISSUE TITLE
Dataset is neither mountable nor unmountable

### ISSUE THREAD

**Issue by ANorwell** [zfsonlinux/zfs#5172]
In a situation where I have ~50K datasets, most of which are unmounted, there seems to be a small probability that datasets get in the following state after creation:
```
# zfs mount array/bfyData/3/31620
filesystem 'array/bfyData/3/31620' is already mounted
cannot mount 'array/bfyData/3/31620': mountpoint or dataset is busy
# zfs unmount array/bfyData/3/31620
cannot unmount 'array/bfyData/3/31620': not currently mounted
```

The dataset does seem to be mounted (`df` shows it as present). The `umount` command successfully unmounts the dataset.

**Comment by DeHackEd** [comment]
See #5029. ZFS checks `/etc/mtab` to determine what and whether something is mounted. This file is managed by user applications and can come out of sync through trickery, negligence or by accident.

You can use `/proc/mounts` as an alternative source for `/etc/mtab` contents to fix it.

**Comment by ANorwell** [comment]
You're right -- the dataset is showing up in `/proc/mounts` but not in `/etc/mtab`. Since it seems like this was already addressed in #5029, I'll close this.
